tests/test_distribution/test_rest_client.py

```diff
@@ -0,0 +1,234 @@
+
+        distribution = DistributionRestClient(mock_client)
+
+        assert distribution.DelegatorWithdrawAddress(
+            QueryDelegatorWithdrawAddressRequest(delegator_address='delegator_addr')) == expected_response
+        assert mock_client.last_base_url == "/cosmos/distribution/v1beta1/delegators/delegator_addr/withdraw_address"
+
+    @staticmethod
+    def test_Params():
+        """Test Params method."""
+        content = {
+            "params": {
+                "community_tax": "0.1",
+                "base_proposer_reward": "0.2",
+                "bonus_proposer_reward": "0.3",
+                "withdraw_addr_enabled": True
+            }
+        }
+        mock_client = MockRestClient(json.dumps(content))
+
+        expected_response = ParseDict(content, QueryParamsResponse())
+
+        distribution = DistributionRestClient(mock_client)
+
+        assert distribution.Params(QueryParamsRequest()) == expected_response
+        assert mock_client.last_base_url == "/cosmos/distribution/v1beta1/params"
+
+    @staticmethod
+    def test_ValidatorCommission():
+        """Test ValidatorCommission method."""
+        content = {
+            "commission": {
+                "commission": [
+                    {
+                        "denom": "string",
+                        "amount": "1234"
+                    }
+                ]
+            }
+        }
+        mock_client = MockRestClient(json.dumps(content))
+
+        expected_response = ParseDict(content, QueryValidatorCommissionResponse())
+
+        distribution = DistributionRestClient(mock_client)
+
+        assert distribution.ValidatorCommission(
+            QueryValidatorCommissionRequest(validator_address='validator_addr')) == expected_response
+        assert mock_client.last_base_url == "/cosmos/distribution/v1beta1/validators/validator_addr/commission"
+
+    @staticmethod
+    def test_ValidatorOutstandingRewards():
+        """Test ValidatorOutstandingRewards method."""
+        content = {
+            "rewards": {
+                "rewards": [
+                    {
+                        "denom": "string",
+                        "amount": "1234"
+                    }
+                ]
+            }
+        }
+        mock_client = MockRestClient(json.dumps(content))
+
+        expected_response = ParseDict(content, QueryValidatorOutstandingRewardsResponse())
+
+        distribution = DistributionRestClient(mock_client)
+
+        assert distribution.ValidatorOutstandingRewards(
+            QueryValidatorOutstandingRewardsRequest(validator_address='validator_addr')) == expected_response
+        assert mock_client.last_base_url == "/cosmos/distribution/v1beta1/validators/validator_addr/outstanding_rewards"
+
+    @staticmethod
+    def test_ValidatorSlashes():
+        """Test ValidatorSlashes method."""
+        content = {
+            "slashes": [],
```

Comment:
would be nice to have at least on slash listed as in  rEST API example

Reply:
I'm not able to query slashes with anything in the list.
I think this endpoint is broken.